src/coreclr/vm/threads.cpp

```diff
@@ -6134,21 +6134,22 @@ HRESULT Thread::CLRSetThreadStackGuarantee(SetThreadStackGuaranteeScope fScope)
         //
         EXTRA_PAGES = 3;
         INDEBUG(EXTRA_PAGES += 1);
+#else // HOST_64BIT
+#ifdef _DEBUG
+        uGuardSize += (1 * GetOsPageSize());    // one extra page for debug infrastructure
+#endif // _DEBUG
+#endif // HOST_64BIT
 
         int ThreadGuardPages = CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_ThreadGuardPages);
```

Comment:
A nit - the GetConfigValue has an optional argument with default value. So this can be simplified to (I've removed the comment for clarity):
```c++
        ULONG uGuardSize = SIZEOF_DEFAULT_STACK_GUARANTEE;
        int   EXTRA_PAGES = 0;
#if defined(HOST_64BIT)
        EXTRA_PAGES = 3;
#endif // HOST_64BIT
        INDEBUG(EXTRA_PAGES += 1);
        int ThreadGuardPages = CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_ThreadGuardPages, EXTRA_PAGES);
        uGuardSize += (ThreadGuardPages * GetOsPageSize());
```